### PR TITLE
Add a feature to set compass target

### PR DESCRIPTION
**Pull request by swooboo**
This PR adds the functionality for setting a compass target on a map.

Also, it sets the compass target of Town to point to dungeon entrance
instead of the town shop.

I did not take level scripts into account, but this will probably help.

### CODE REVIEW

**Comment by swooboo** [comment]
What are these 4 issues introduced that codebeat reports?

**Comment by swooboo** [comment]
Accidentally pushed the changed PixelDungeon.iml - I will re-submit the pull request without the change, correct? Or is it not critical?

**Comment by swooboo** [comment]
I think it's safe to add the `PixelDungeon.iml` to `.gitignore`, since it's generated from the Gradle files. See answers here - https://stackoverflow.com/questions/30737082/what-are-iml-files-in-android-studio . (It shouldn't be tracked with version control)

**Comment by swooboo** [comment]
Alright, I've removed the `.iml` file from my commits and also fixed a small bug and a small typo. Should be better now. We need to consider adding it to `.gitignore`.

File: PixelDungeon/src/main/java/com/watabou/pixeldungeon/levels/Level.java [hunk 1]
```diff
@@ -118,6 +118,20 @@ public boolean hasExit(Integer index) {
...
+		if (hasCompassTarget()) {
+			return compassTarget;
+		}
+
+		throw new TrackedRuntimeException("no compass target, current value of compassTarget: " + compassTarget);
+	}
+
+	// Check whether the level has a compass target
+	public boolean hasCompassTarget() {
+		return -1 != compassTarget;
```
Comment (Mikhael-Danilov) [inline PixelDungeon/src/main/java/com/watabou/pixeldungeon/levels/Level.java]: Please use cellValid method here.

File: PixelDungeon/src/main/java/com/watabou/pixeldungeon/ui/Compass.java [hunk 2]
```diff
@@ -39,10 +39,20 @@ public Compass( int cell ) {
...
+
+	public void setCell(int cell) {
+		this.cell = cell;
+		cellCenter = DungeonTilemap.tileCenterToWorld( cell );	// Exact location of the center of the tile
+	}
+
+	// Set the cell with x,y coordinates
+	public void setCell( int x, int y) {
+		int width = Dungeon.level.getWidth();
+		setCell( y*width + x );	// Since x = cell%width and y = cell/width
```
Comment (Mikhael-Danilov) [inline PixelDungeon/src/main/java/com/watabou/pixeldungeon/ui/Compass.java]: Please use Level.cell method instead of direct calculations here.

**Comment by Mikhael-Danilov** [comment]
Thanks! 
Take a look at my comments, please, and then I will merge this PR.

**Comment by Mikhael-Danilov** [comment]
For some starnge reason codebeat do not want to provide details...

https://codebeat.co/projects/github-com-nyrds-pixel-dungeon-remix-master/pull_requests

**Comment by swooboo** [comment]
@Mikhael-Danilov done.

**Pull request merged**